services/system-user-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/keycloak/KeycloakUserDataAccessLayer.java

```diff
@@ -0,0 +1,157 @@
+        // Create new users with a temp password that must be changed on first sign in
+        CredentialRepresentation tempPassword = new CredentialRepresentation();
+        tempPassword.setType("password");
+        tempPassword.setTemporary(Boolean.TRUE);
+        tempPassword.setValue(Utils.randomString(12));
+
+        UserRepresentation keycloakUser = toKeycloakUser(user);
+        keycloakUser.setCredentials(List.of(tempPassword));
+        keycloakUser.setRequiredActions(List.of("UPDATE_PASSWORD"));
+        keycloakUser.setEnabled(Boolean.TRUE);
+        keycloakUser.setGroups(List.of(keycloak.getAdminGroupPath(event)));
+
+        UserRepresentation createdUser = keycloak.createUser(event, keycloakUser);
+        return toSystemUser(createdUser);
+    }
+
+    @Override
+    public void deleteUser(Map<String, Object> event, String username) {
+        keycloak.deleteUser(event, username);
+    }
+
+    // VisibleForTesting
+    static SystemUser toSystemUser(UserRepresentation keycloakUser) {
+        SystemUser user = null;
+        if (keycloakUser != null) {
+            user = new SystemUser();
+            user.setId(keycloakUser.getId());
+            user.setCreated(LocalDateTime.ofInstant(
+                    Instant.ofEpochMilli(keycloakUser.getCreatedTimestamp()),
+                    ZoneId.of("UTC")));
+            // Keycloak doesn't track when a user was last modified
+            user.setModified(null);
+            user.setActive(keycloakUser.isEnabled());
+            user.setUsername(keycloakUser.getUsername());
+            user.setFirstName(keycloakUser.getFirstName());
+            user.setLastName(keycloakUser.getLastName());
+            user.setEmail(keycloakUser.getEmail());
+            user.setEmailVerified(keycloakUser.isEmailVerified());
+            if (!keycloakUser.getRequiredActions().isEmpty()) {
+                user.setStatus(keycloakUser.getRequiredActions().get(0));
+            }
+        }
+        return user;
+    }
+
+    // VisibleForTesting
+    static UserRepresentation toKeycloakUser(SystemUser user) {
+        UserRepresentation keycloakUser = null;
+        if (user != null) {
+            keycloakUser = updateUserRepresentation(user, new UserRepresentation());
+        }
+        return keycloakUser;
+    }
+
+    // VisibleForTesting
+    static UserRepresentation updateUserRepresentation(SystemUser user, UserRepresentation keycloakUser) {
+        if (user != null) {
+            keycloakUser.setUsername(user.getUsername());
+            keycloakUser.setFirstName(user.getFirstName());
+            keycloakUser.setLastName(user.getLastName());
+            keycloakUser.setEnabled(user.getActive());
+            keycloakUser.setEmail(user.getEmail());
+            keycloakUser.setEmailVerified(Boolean.TRUE.equals(user.getEmailVerified()));
+            keycloakUser.setCreatedTimestamp(LocalDateTime.now().toInstant(ZoneOffset.UTC).toEpochMilli());
```

Comment:
Now that we have update implemented, should we wrap this in an if/else to see if our incoming SystemUser object already has a created timestamp -- so we don't turn created into modified.
```java
if (user.getCreated() == null) {
    keycloakUser.setCreatedTimestamp(LocalDateTime.now().toInstant(ZoneOffset.UTC).toEpochMilli());
} else {
    keycloakUser.setCreatedTimestamp(user.getCreated().toInstant(ZoneOffset.UTC).toEpochMilli());
}
```